diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/arch/arm/mach-mx3/Makefile linux-2.6.7-mxc1.5-gcov.new/arch/arm/mach-mx3/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/arch/arm/mach-mx3/Makefile	2004-11-04 11:10:46.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/arch/arm/mach-mx3/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,14 @@
 # Makefile for the linux kernel.
 #
 
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+
 # Object file lists.
 
 obj-y			:= cpu.o irq.o mm.o sdma.o gpio_common.o time.o spba.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/arch/mach-mxc91131/Makefile linux-2.6.7-mxc1.5-gcov.new/arch/arm/mach-mxc91131/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/arch/arm/mach-mxc91131/Makefile	2004-11-04 11:10:46.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/arch/arm/mach-mxc91131/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -2,7 +2,15 @@
 # Makefile for the linux kernel.
 #
 
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+
 # Object file lists.

 obj-y                   := devices.o iomux.o system.o cpu.o mm.o time.o clock.o serial.o dma.o
 obj-$(CONFIG_MACH_MXC91131EVB)  += mxc91131evb.o mxc91131evb_gpio.o mxc_pm.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/arch/mach-mxc91231/Makefile linux-2.6.7-mxc1.5-gcov.new/arch/arm/mach-mxc91231/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/arch/arm/mach-mxc91231/Makefile       2004-11-04 11:10:46.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/arch/arm/mach-mxc91231/Makefile 2004-11-04 12:25:12.000000000 +0100
@@ -2,7 +2,15 @@
 # Makefile for the linux kernel.
 #

-# Object file lists.
+
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+

 obj-y                   := devices.o iomux.o system.o cpu.o mm.o time.o clock.o serial.o dma.o
 obj-$(CONFIG_MACH_MXC27530EVB)  += mxc27530evb.o mxc27530evb_gpio.o mxc_pm.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/arch/arm/mach-mxc91321/Makefile linux-2.6.7-mxc1.5-gcov.new/arch/arm/mach-mxc91321/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/arch/arm/mach-mxc91321/Makefile       2004-11-04 11:10:46.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/arch/arm/mach-mxc91321/Makefile 2004-11-04 12:25:12.000000000 +0100
@@ -2,7 +2,15 @@
 # Makefile for the linux kernel.
 #
 #


-# Object file lists.
+
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+
 
 obj-y                                   := devices.o iomux.o system.o cpu.o mm.o time.o clock.o serial.o dma.o
 obj-$(CONFIG_MACH_I30030EVB)            += i30030evb.o i30030evb_gpio.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/char/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/char/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/char/Makefile	2004-11-04 11:11:14.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/char/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -2,7 +2,15 @@
 #
 # Makefile for the kernel character device drivers.
 #
 
 #
 # This file contains the font map for the default (hardware) font
 #
+
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_mu.o += $(GCOV_FLAGS)
+CFLAGS_mxc_sdma_tty.o += $(GCOV_FLAGS)
+CFLAGS_mxc_ipc.o += $(GCOV_FLAGS)
+CFLAGS_mxc_rtc.o += $(GCOV_FLAGS)
+endif
+
 FONTMAPFILE = cp437.uni
 
 obj-y    += mem.o random.o tty_io.o n_tty.o tty_ioctl.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/char/watchdog/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/char/watchdog/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/char/watchdog/Makefile 2004-11-04 11:11:14.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/char/watchdog/Makefile   2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,15 @@
 #
 # Makefile for the WatchDog device drivers.
 #

+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_wdt.o += $(GCOV_FLAGS)
+endif
+
 # Only one watchdog can succeed. We probe the hardware watchdog
 # drivers first, then the softdog driver.  This means if your hardware
 # watchdog dies or is 'borrowed' for some reason the software watchdog
 # still gives you some cover.
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/ide/arm/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/ide/arm/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/ide/arm/Makefile 2004-11-04 11:11:14.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/ide/arm/Makefile   2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,15 @@
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_ide.o += $(GCOV_FLAGS)
+endif
+
 obj-$(CONFIG_BLK_DEV_IDE_ICSIDE)        += icside.o
 obj-$(CONFIG_BLK_DEV_IDE_RAPIDE)        += rapide.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/input/keyboard/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/input/keyboard/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/input/keyboard/Makefile 2004-11-04 11:11:14.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/input/keyboard/Makefile   2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,15 @@
 #
 # Makefile for the input core drivers.
 #
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_keyb.o += $(GCOV_FLAGS)
+endif
+
 # Each configuration option enables a list of files.

 obj-$(CONFIG_KEYBOARD_ATKBD)            += atkbd.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/gcov/Kconfig linux-2.6.7-mxc1.5-gcov.new/drivers/gcov/Kconfig
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/gcov/Kconfig	2004-11-04 11:13:09.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/gcov/Kconfig	2004-11-04 12:25:12.000000000 +0100
@@ -27,6 +27,13 @@
 	If you say Y here, it will compile the entire kernel with coverage
 	option enabled.
 
+config GCOV_FREESCALE_ALL
+	bool "Profile all freescale drivers"
+	depends on GCOV_PROFILE
+	---help---
+	If you say Y here, it will compile all the Freescale drivers with coverage
+	option enabled.
+	
 config GCOV_PROC
 	tristate "Provide GCOV proc file system entry"
 	depends on GCOV_PROFILE && PROC_FS
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/i2c/busses/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/i2c/busses/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/i2c/busses/Makefile	2004-11-04 11:11:11.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/i2c/busses/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,13 @@
 # Makefile for the i2c bus drivers.
 #
 
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_i2c.o += $(GCOV_FLAGS)
+endif
+
 obj-$(CONFIG_I2C_ALI1535)	+= i2c-ali1535.o
 obj-$(CONFIG_I2C_ALI1563)	+= i2c-ali1563.o
 obj-$(CONFIG_I2C_ALI15X3)	+= i2c-ali15x3.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/i2c/chips/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/i2c/chips/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/i2c/chips/Makefile	2004-11-04 11:11:11.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/i2c/chips/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,13 @@
 # Makefile for the kernel hardware sensors chip drivers.
 #
 
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_i2c_client.o += $(GCOV_FLAGS)
+endif
+
 # asb100, then w83781d go first, as they can override other drivers' addresses.
 obj-$(CONFIG_SENSORS_ASB100)	+= asb100.o
 obj-$(CONFIG_SENSORS_W83627HF)	+= w83627hf.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mmc/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mmc/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mmc/Makefile	2004-11-04 11:11:15.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mmc/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,13 @@
 # Makefile for the kernel mmc device drivers.
 #
 
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_mmc.o += $(GCOV_FLAGS)
+endif
+
 #
 # Core
 #
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/media/video/mxc/capture/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/media/video/mxc/capture/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/media/video/mxc/capture/Makefile  2004-11-04 11:11:15.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/media/video/mxc/capture/Makefile    2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,13 @@
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_v4l2_capture.o += $(GCOV_FLAGS)
+endif
+
 ifeq ($(CONFIG_VIDEO_MXC_IPU_CAMERA),y)
        obj-$(CONFIG_VIDEO_MXC_CAMERA) += mxc_v4l2_capture.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/media/video/mxc/output/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/media/video/mxc/output/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/media/video/mxc/output/Makefile  2004-11-04 11:11:15.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/media/video/mxc/output/Makefile    2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,13 @@
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_v4l2_output.o += $(GCOV_FLAGS)
+endif
+
 ifeq ($(CONFIG_VIDEO_MXC_EMMA_OUTPUT),y)
        mx27_output-objs := mx27_v4l2_output.o mx27_pp.o
        obj-$(CONFIG_VIDEO_MXC_OUTPUT) += mx27_output.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mtd/maps/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mtd/maps/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mtd/maps/Makefile	2004-11-04 11:11:25.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mtd/maps/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -3,6 +3,16 @@
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_nor.o += $(GCOV_FLAGS)
+endif
+
 ifeq ($(CONFIG_MTD_COMPLEX_MAPPINGS),y)
 obj-$(CONFIG_MTD)		+= map_funcs.o
 endif
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mtd/nand/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mtd/nand/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mtd/nand/Makefile	2004-11-04 11:11:26.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mtd/nand/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -3,6 +3,14 @@
#
# linux/drivers/nand/Makefile
#
# $Id: linux-L26-21-9-gcov-freescale.patch,v 1.1.1.1 2008/04/14 09:02:59 b06080 Exp $

+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(GCOV_FREESCALE_ALL), y)
+CFLAGS_mxc_nd.o += $(GCOV_FLAGS)
+endif
+

 obj-$(CONFIG_MTD_NAND)                  += nand.o nand_ecc.o
 obj-$(CONFIG_MTD_NAND_IDS)              += nand_ids.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/hmp4e/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/hmp4e/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/hmp4e/Makefile	2004-11-04 11:11:30.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/hmp4e/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -2,5 +2,13 @@
 #
 # Makefile for the MPEG4 Encoder kernel module.

+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+CFLAGS_mxc_hmp4e.o += $(GCOV_FLAGS)
+endif
+

 obj-$(CONFIG_MXC_HMP4E)                 += mxc_hmp4e.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/ipu/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/ipu/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/ipu/Makefile	2004-11-04 11:11:30.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/ipu/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -1,2 +1,9 @@
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+

 obj-$(CONFIG_MXC_IPU_PF)        +=   pf/
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/ipu/pf/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/ipu/pf/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/ipu/pf/Makefile	2004-11-04 11:11:30.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/ipu/pf/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -1,3 +1,10 @@
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+
 obj-$(CONFIG_MXC_IPU_PF) += mxc_pf.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/security/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/security/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/security/Makefile	2004-11-04 11:11:30.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/security/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -2,5 +2,14 @@
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+CFLAGS_mxc_scc.o += $(GCOV_FLAGS)
+CFLAGS_mxc_rtic.o += $(GCOV_FLAGS)
+CFLAGS_mxc_hacc.o += $(GCOV_FLAGS)
+CFLAGS_mxc_sec_mod.o += $(GCOV_FLAGS)
+endif
+
 # Makefile for the Linux MXC Security API
 obj-$(CONFIG_MXC_SECURITY_SCC) += mxc_scc.o
 obj-$(CONFIG_MXC_SECURITY_RTIC) += mxc_rtic.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/security/rng/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/security/rng/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/security/rng/Makefile 2004-11-04 11:11:30.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/security/rng/Makefile   2004-11-04 12:25:12.000000000 +0100
@@ -2,5 +2,14 @@
 # Makefile for the Linux RNG Driver
 #
 # This makefile works within a kernel driver tree

 # Makefile for rng_driver
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+
 # Possible configurable paramters
 CFG_RNGA += -DRNGA_MAX_REQUEST_SIZE=32
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/security/sahara2/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/security/sahara2/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/security/sahara2/Makefile 2004-11-04 11:11:30.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/security/sahara2/Makefile   2004-11-04 12:25:12.000000000 +0100
@@ -2,5 +2,14 @@
 # Makefile for the Linux Sahara2 driver
 #
 # This makefile works within a kernel driver tree
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+
 # Need to augment this to support optionally building user-mode support
API_SOURCES = fsl_shw_sym.c  fsl_shw_user.c  fsl_shw_hash.c  fsl_shw_auth.c \
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/spi/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/spi/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/spi/Makefile	2004-11-04 11:11:30.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/spi/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -1,6 +1,14 @@
 #
 # Makefile for the kernel SPI device drivers.
 
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+
 #ifneq($(KERNELRELEASE))
 
 obj-$(CONFIG_SPI) 		+= spimod.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/spi/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/spi/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/spi/Makefile      2004-11-04 11:11:30.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/spi/Makefile        2004-11-04 12:25:12.000000000 +0100
@@ -1,6 +1,14 @@
 #
 # Makefile for kernel SPI drivers.
 #
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+CFLAGS_mxc_spi.o += $(GCOV_FLAGS)
+endif
+
 ifeq ($(CONFIG_SPI_DEBUG),y)
 EXTRA_CFLAGS += -DDEBUG
 endif
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/ssi/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/ssi/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/ssi/Makefile	2004-11-04 11:11:30.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/ssi/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -1,6 +1,14 @@
 #
 # Makefile for the kernel SSI device drivers.
 
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+
 
 obj-$(CONFIG_SSI) 		+= ssimod.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/net/irda/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/net/irda/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/net/irda/Makefile     2004-11-04 11:11:04.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/net/irda/Makefile       2004-11-04 12:25:12.000000000 +0100
@@ -4,6 +4,13 @@
 # 9 Aug 2000, Christoph Hellwig <hch@infradead.org>
 # Rewritten to use lists instead of if-statements.
 #
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_fir.o += $(GCOV_FLAGS)
+endif

 # Old SIR drivers
 obj-$(CONFIG_IRPORT_SIR)       +=              irport.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/vpu/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/vpu/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/mxc/vpu/Makefile	2004-11-04 11:11:04.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/mxc/vpu/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -4,6 +4,13 @@
 # Makefile for the Linux VPU drivers.
 # 25 May 2006

+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_vpu.o += $(GCOV_FLAGS)
+endif

 # Each configuration option enables a list of files.

 obj-$(CONFIG_MXC_VPU)                  += mxc_vpu.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/net/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/net/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/net/Makefile	2004-11-04 11:11:06.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/net/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -2,6 +2,14 @@
 # Makefile for the Linux network (ethercard) device drivers.
 #
 
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+CFLAGS_3c59x.o += $(GCOV_FLAGS)
+endif
+
 ifeq ($(CONFIG_ISDN_PPP),y)
   obj-$(CONFIG_ISDN) += slhc.o
 endif
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/serial/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/serial/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/serial/Makefile	2004-11-04 11:11:17.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/serial/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -3,6 +3,12 @@
 #
 #  $Id: linux-L26-21-9-gcov-freescale.patch,v 1.1.1.1 2008/04/14 09:02:59 b06080 Exp $
 #
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL),y)
+CFLAGS_mxc_uart.o += $(GCOV_FLAGS)
+endif
 
 serial-8250-y :=
 serial-8250-$(CONFIG_SERIAL_8250_ACPI) += 8250_acpi.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/video/mxc/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/video/mxc/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/video/mxc/Makefile	2004-11-04 11:11:10.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/video/mxc/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -1 +1,9 @@
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+
 obj-$(CONFIG_FB_MXC)        += mxcfb.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/otg/ocd/mxc//Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/otg/ocd/mxc//Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/otg/ocd/mxc//Makefile    2004-11-04 11:11:10.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/otg/ocd/mxc//Makefile      2004-11-04 12:25:12.000000000 +0100
@@ -1 +1,9 @@
 #
 # Makefile for the kernel USBD (device not host) drivers.
 # @(#) sl@belcarra.com|otg/ocd/mxc/Makefile-l26|20060215215144|37795
 #
 # Copyright (c) 2005 Belcarra Technologies Corp

+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+EXTRA_CFLAGS += $(GCOV_FLAGS)
+endif
+

 extra-y := mxc-gptcr.o mxc-ocd.o mxc-pcd.o mxc-procfs.o mxc-l26.o mxc-hcd.o
 ifdef CONFIG_ARCH_MXC91321
 extra-y += mxc91321-gpio.o mc13783.o
 endif
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/w1/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/w1/Makefile
-- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/w1/Makefile    2004-11-04 11:11:10.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/w1/Makefile      2004-11-04 12:25:12.000000000 +0100
@@ -1 +1,9 @@
 #
 # Makefile for the Dallas's 1-wire bus.
 #

+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+CFLAGS_mxc_w1.o += $(GCOV_FLAGS)
+endif
+

 ifneq ($(CONFIG_NET), y)
 EXTRA_CFLAGS    += -DNETLINK_DISABLED
 endif
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/pcmcia/Makefile linux-2.6.7-mxc1.5-gcov.new/drivers/pcmcia/Makefile
-- linux-2.6.7-mxc1.5-gcov-2.6.5/drivers/pcmcia/Makefile    2004-11-04 11:11:10.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/drivers/pcmcia/Makefile      2004-11-04 12:25:12.000000000 +0100
@@ -1 +1,9 @@
 #
 # Makefile for the kernel pcmcia subsystem (c/o David Hinds)
 #

+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+CFLAGS_mx31ads-pcmcia.o += $(GCOV_FLAGS)
+endif
+

 ifeq ($(CONFIG_PCMCIA_DEBUG),y)
 EXTRA_CFLAGS += -DDEBUG
 endif

diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/sound/oss/Makefile linux-2.6.7-mxc1.5-gcov.new/sound/oss/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/sound/oss/Makefile	2004-11-04 11:11:39.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/sound/oss/Makefile	2004-11-04 12:25:12.000000000 +0100
@@ -3,6 +3,15 @@
 # 18 Apr 1998, Michael Elizabeth Chastain, <mailto:mec@shout.net>
 # Rewritten to use lists instead of if-statements.
 
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+# 
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+CFLAGS_mxc-audio.o += $(GCOV_FLAGS)
+CFLAGS_mxc-hwdep.o += $(GCOV_FLAGS)
+endif
+
 # Each configuration option enables a list of files.
 
 obj-$(CONFIG_SOUND_OSS)		+= sound.o
diff -Naur linux-2.6.7-mxc1.5-gcov-2.6.5/sound/arm/Makefile linux-2.6.7-mxc1.5-gcov.new/sound/arm/Makefile
--- linux-2.6.7-mxc1.5-gcov-2.6.5/sound/arm/Makefile    2004-11-04 11:11:39.000000000 +0100
+++ linux-2.6.7-mxc1.5-gcov.new/sound/arm/Makefile      2004-11-04 12:25:12.000000000 +0100
@@ -3,6 +3,15 @@
 #
 # Makefile for ALSA
 #

 #
 # Define the header file locations for PMIC drivers.
 #
+#
+# cgag1c@motorola.com [TLS FR] 02/06/2004
+# test for gcov
+#
+ifeq ($(CONFIG_GCOV_FREESCALE_ALL), y)
+CFLAGS_mxc-alsa-pmic.o += $(GCOV_FLAGS)
+CFLAGS_mxc-alsa-mixer.o += $(GCOV_FLAGS)
+CFLAGS_snd-mxc-alsa-pmic.o += $(GCOV_FLAGS)
+CFLAGS_snd-mxc-alsa-mixer.o += $(GCOV_FLAGS)
+endif
+
 ifeq ($(CONFIG_MXC_MC13783_LEGACY),y)
 CFLAGS_mxc-alsa-mc13783.o  = -I$(TOPDIR)/drivers/mxc
 endif
